### PR TITLE
refactor: move go new app template into core

### PR DESCRIPTION
**Pull request by jmar910**
## Description of change

Fixes https://github.com/meroxa/turbine-go/issues/133

## Type of change

- [ ]  New feature
- [ ]  Bug fix
- [x]  Refactor
- [ ]  Documentation

## How was this tested?

- [x]  Unit Tests
- [ ]  Tested in staging
- [ ]  Tested in minikube

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -7,23 +7,33 @@ require (
 	github.com/google/uuid v1.3.0
 	github.com/heimdalr/dag v1.2.1
 	github.com/jedib0t/go-pretty/v6 v6.4.6
+	github.com/meroxa/turbine-go v1.0.0
```
Comment (raulb) [inline go.mod:10]: I think it's fine to point to 1.0.0, but something worth calling out is that we have added more things after: https://github.com/meroxa/turbine-go/compare/e6470e58b719a02c4314da4e512afef7078c4fe8...main.

Maybe we could:

1. Create a new tagged release in turbine-go and update this one 
2. Update to a specific commit hash.

@jmar910 what do you think?

File: Makefile [hunk 2]
```diff
@@ -8,7 +8,10 @@ vet:
 
 .PHONY: test
 test: vet
-	go test ./... -coverprofile=c.out -covermode=atomic -v
+	go test `go list ./... | grep -v 'turbine-core\/pkg\/app\/templates\/go'` \
+	-coverprofile=c.out \
+	-covermode=atomic \
+	-v
```
Comment (raulb) [inline Makefile:14]: Nice

File: pkg/app/init_test.go [hunk 3]
```diff
@@ -27,6 +27,15 @@ func TestAppInit_createAppDirectory(t *testing.T) {
 			},
 			wantErr: false,
 		},
+		{
+			name: "creates the app directory",
+			fields: fields{
+				AppName:  "createappdir",
+				Language: "go",
```
Comment (raulb) [inline pkg/app/init_test.go:34]: Now that we're adding more languages, maybe it's worth considering unifying what name we're using to specify which language is to make sure we don't run into cases like we send from the CLI "golang" and turbine-core expects "go". This seems to be the case [now](https://github.com/meroxa/cli/blob/bc64b8621a260380d8a7e8f336bffbcadbce19e2/cmd/meroxa/turbine/utils.go#L22).

Suggestion, move these to turbine-core and rely on those everywhere: https://github.com/meroxa/cli/blob/bc64b8621a260380d8a7e8f336bffbcadbce19e2/cmd/meroxa/turbine/utils.go#L22-L27.

@jmar910 what do you think?

File: pkg/app/init_test.go [hunk 4]
```diff
@@ -122,7 +155,7 @@ func TestAppInit_duplicateFile(t *testing.T) {
 			if err := a.duplicateFile(tt.args.fileName); (err != nil) != tt.wantErr {
 				t.Errorf("AppInit.duplicateFile() error = %v, wantErr %v", err, tt.wantErr)
 			}
-			require.FileExists(t, filepath.Join(tt.fields.Path, tt.fields.AppName, "Gemfile"))
+			require.FileExists(t, filepath.Join(tt.fields.Path, tt.fields.AppName, tt.args.fileName))
```
Comment (raulb) [inline pkg/app/init_test.go:158]: Good change. It seems like we should have used that anyways before.

File: pkg/app/init_test.go [hunk 5]
```diff
@@ -203,10 +261,10 @@ func TestAppInit_Init(t *testing.T) {
 			}
 
 			require.DirExists(t, filepath.Join(tt.fields.Path, tt.fields.AppName))
-			require.FileExists(t, filepath.Join(tt.fields.Path, tt.fields.AppName, "app.json"))
-			require.FileExists(t, filepath.Join(tt.fields.Path, tt.fields.AppName, "app.rb"))
-			require.FileExists(t, filepath.Join(tt.fields.Path, tt.fields.AppName, "Gemfile"))
-			require.FileExists(t, filepath.Join(tt.fields.Path, tt.fields.AppName, "fixtures", "demo.json"))
+			for _, file := range tt.wantFiles {
```
Comment (raulb) [inline pkg/app/init_test.go:264]: "file" is a common type name in the standard library, so maybe we could rename this one to make it more explicit. 

Something like `wantFile` for example.

Comment (raulb) [inline pkg/app/init_test.go:264]:
```suggestion
			for _, wantFile := range tt.wantFiles {
```

File: pkg/app/init_test.go [hunk 6]
```diff
@@ -203,10 +261,10 @@ func TestAppInit_Init(t *testing.T) {
...
+				require.FileExists(t, filepath.Join(tt.fields.Path, tt.fields.AppName, file))
```
Comment (raulb) [inline pkg/app/init_test.go:265]:
```suggestion
				require.FileExists(t, filepath.Join(tt.fields.Path, tt.fields.AppName, wantFile))
```

File: pkg/app/templates/go/.gitignore [hunk 7]
```diff
@@ -0,0 +1,2 @@
+{{.AppName}}
+{{.AppName}}.cross
```
Comment (raulb) [inline pkg/app/templates/go/.gitignore:2]: Add empty line here.

File: pkg/app/templates/go/README.md [hunk 8]
```diff
@@ -0,0 +1,221 @@
...
+* **Native Developer Tooling:** Turbine doesn't come with any bespoke DSL or patterns. Write software like you normally would!
+
+* **Fits into Existing DevOps Workflows:** Build, test, and deploy. Turbine encourages best practices from the start. Don't test your data app in production ever again.
+
+* **Local Development mirrors Production:** When running locally, you'll immediately see how your app reacts to data. What you get there will be exactly what happens in production but with _scale_ and _speed_.
+
+* **Available in many different programming langauages:** Turbine started out in Go but is available in other languages too:
+    * [Go](https://github.com/meroxa/turbine-go)
+    * [Javascript](https://github.com/meroxa/turbine-js)
+    * [Python](https://github.com/meroxa/turbine-py)
```
Comment (raulb) [inline pkg/app/templates/go/README.md:22]: Not part of this PR, but we should track somewhere that we go over these readme and rely on something more automated and up to date (e.g.: Ruby is not there)

File: pkg/app/templates/go/app.json [hunk 9]
```diff
@@ -0,0 +1,8 @@
+{
+  "name": "{{.AppName}}",
+  "language": "golang",
+  "environment": "common",
+  "resources": {
+    "source_name": "fixtures/demo-cdc.json"
+  }
+}
```
Comment (raulb) [inline pkg/app/templates/go/app.json:8]: Add empty line here.

File: pkg/app/templates/go/fixtures/demo-cdc.json [hunk 10]
```diff
@@ -0,0 +1,358 @@
...
+            "xmin": null
+          },
+          "op": "r",
+          "ts_ms": 1644362356743,
+          "transaction": null
+        }
+      }
+    }
+  ]
+}
```
Comment (raulb) [inline pkg/app/templates/go/fixtures/demo-cdc.json:358]: Add empty line here.

File: pkg/app/templates/go/fixtures/demo-no-cdc.json [hunk 11]
```diff
@@ -0,0 +1,166 @@
...
+          "shipping_address": "43 Sunnyslope Street Port Richey, FL 34668",
+          "customer_email": "userc@example.com",
+          "updated_at": "2022-01-22 19:43:12",
+          "created_at": "2022-01-22 19:43:12",
+          "deleted_at": null
+        }
+      }
+    }
+  ]
+}
```
Comment (raulb) [inline pkg/app/templates/go/fixtures/demo-no-cdc.json:166]: add empty line.

**Review by raulb — CHANGES_REQUESTED** [review]
@jmar910 Overall looks good. Just some comments and requested changes. I wonder whether we should vendor those packages at all. If we're considering bringing other languages over, my inclination is to leave this as light as possible and not vendor unless is strictly necessary (which it doesn't seem the case yet).

File: pkg/app/init_test.go (hunk 3 above)
Comment (raulb) [reply pkg/app/init_test.go:34]: This is currently "breaking" my other [PR here](https://github.com/meroxa/cli/pull/699), so I'll go ahead and propose something in turbine-core for a review.

Comment (raulb) [reply pkg/app/init_test.go:34]: Got this PR already https://github.com/meroxa/turbine-core/pull/110.

Comment (raulb) [reply pkg/app/init_test.go:34]: This is how it'd look like in CLI https://github.com/meroxa/cli/pull/700.

File: pkg/app/templates/go/README.md (hunk 8 above)
Comment (jmar910) [reply pkg/app/templates/go/README.md]: 👍

**Review by raulb — APPROVED** [review]

File: go.mod (hunk 1 above)
Comment (raulb) [reply go.mod:10]: It sounds like it's perfectly fine.
